mono/metadata/android.h

```diff
@@ -0,0 +1,79 @@
+
+#ifndef __MONO_METADATA_ANDROID_H__
+#define __MONO_METADATA_ANDROID_H__
+
+#include <glib.h>
+
+#include <jni.h>
+
+#include "object.h"
+#include "utils/mono-publib.h"
+
+/* We're implementing getifaddrs behavior, this is the structure we use. It is exactly the same as
+ * struct ifaddrs defined in ifaddrs.h but since bionics doesn't have it we need to mirror it here.
+ */
+struct _monodroid_ifaddrs {
+	struct _monodroid_ifaddrs *ifa_next; /* Pointer to the next structure.      */
+
+	gchar *ifa_name;                      /* Name of this network interface.     */
+	guint ifa_flags;              /* Flags as from SIOCGIFFLAGS ioctl.   */
+
+	struct sockaddr *ifa_addr;           /* Network address of this interface.  */
+	struct sockaddr *ifa_netmask;        /* Netmask of this interface.          */
+	union {
+		/* At most one of the following two is valid.  If the IFF_BROADCAST
+		   bit is set in `ifa_flags', then `ifa_broadaddr' is valid.  If the
+		   IFF_POINTOPOINT bit is set, then `ifa_dstaddr' is valid.
+		   It is never the case that both these bits are set at once.  */
+		struct sockaddr *ifu_broadaddr;  /* Broadcast address of this interface. */
+		struct sockaddr *ifu_dstaddr;    /* Point-to-point destination address.  */
+	} ifa_ifu;
+	/* These very same macros are defined by <net/if.h> for `struct ifaddr'.
+	   So if they are defined already, the existing definitions will be fine.  */
+# ifndef _monodroid_ifa_broadaddr
+#  define _monodroid_ifa_broadaddr ifa_ifu.ifu_broadaddr
+# endif
+# ifndef _monodroid_ifa_dstaddr
+#  define _monodroid_ifa_dstaddr   ifa_ifu.ifu_dstaddr
+# endif
+
+	gpointer ifa_data;               /* Address-specific data (may be unused).  */
+};
+
+MONO_API void
+mono_jvm_initialize (JavaVM *vm);
+
+JNIEnv*
+mono_jvm_get_jnienv (void);
+
+MONO_API gpointer
+_monodroid_timezone_get_default_id (void);
+
+MONO_API gint
+_monodroid_getifaddrs (struct _monodroid_ifaddrs **ifap);
+
+MONO_API void
+_monodroid_freeifaddrs (struct _monodroid_ifaddrs *ifa);
+
+MONO_API void
+_monodroid_detect_cpu_and_architecture (guint16 *built_for_cpu, guint16 *running_on_cpu, guint8 *is64bit);
+
+MONO_API void
+monodroid_add_system_property (const gchar *name, const gchar *value);
+
+MONO_API gint32
+monodroid_get_system_property (const gchar *name, gchar **value);
+
+MONO_API gint32
+_monodroid_get_dns_servers (gpointer *dns_servers_array);
+
+MONO_API MonoBoolean
+_monodroid_get_network_interface_up_state (const gchar *ifname, MonoBoolean *is_up);
```

Comment:
This (and other?) API(s) is *declared*, but it's not *implemented*. In fact, this PR appears to *remove* the only definition for `_monodroid_get_network_interface_up_state()` (in `sdks/android/jni/runtime-bootstrap.c`).

I am thus *very* confused about what you are attempting to do.

Reply:
I *hate* it when GitHub *hides* diffs because they're "too big."

Reply:
I don't like the github behavior either.